pkg/processor/cascadingfilterprocessor/sampling/drop_trace_factory.go

```diff
@@ -55,11 +57,16 @@ func NewDropTraceEvaluator(logger *zap.Logger, cfg config.TraceRejectCfg) (DropT
 		}
 	}
 
+	if cfg.StatusCode != nil {
+		statusCode = cfg.StatusCode
+	}
+
 	return &dropTraceEvaluator{
 		stringAttr:  stringAttrFilter,
 		numericAttr: numericAttrFilter,
 		attrs:       attrsFilter,
 		operationRe: operationRe,
+		statusCode:  statusCode,
```

Comment:
I would change it to the following as it sets `nil` if `cfg.StatusCode` is `nil` otherwise the value

```suggestion
		statusCode:  cfg.StatusCode,
```

No need for:

```
var statusCode *string // it sets statusCode to nil

// It sets statusCode to the value if cfg.StatusCode is not nil
if cfg.StatusCode != nil {
  statusCode = cfg.StatusCode
}
```

Reply:
done!